docs/pages/2020_Intro_Week/sections/workflows_adv.rst

```diff
@@ -287,30 +332,25 @@ To "register" a process handler for a base restart work chain implementation, yo
 
 .. code-block:: python
 
-    from aiida.engine import BaseRestartWorkChain, process_handler
-
-    class ArithmeticAddBaseWorkChain(BaseRestartWorkChain):
-
-        _process_class = ArithmeticAddCalculation
-
-        ...
+    from aiida.engine import process_handler, ProcessHandlerReport
 
-        @process_handler
-        def handle_negative_sum(self, node):
-            """Check if the calculation failed with `ERROR_NEGATIVE_NUMBER`.
+    @process_handler
+    def handle_negative_sum(self, node):
+        """Check if the calculation failed with `ERROR_NEGATIVE_NUMBER`.
 
-            If this is the case, simply make the inputs positive by taking the absolute value.
+        If this is the case, simply make the inputs positive by taking the absolute value.
 
-            :param node: the node of the subprocess that was ran in the current iteration.
-            :return: optional :class:`~aiida.engine.processes.workchains.utils.ProcessHandlerReport` instance to signal
-                that a problem was detected and potentially handled.
-            """
-            if node.exit_status == ArithmeticAddCalculation.exit_codes.ERROR_NEGATIVE_NUMBER.status:
-                self.ctx.inputs['x'] = Int(abs(node.inputs.x.value))
-                self.ctx.inputs['y'] = Int(abs(node.inputs.y.value))
-                return ProcessHandlerReport()
+        :param node: the node of the subprocess that was ran in the current iteration.
+        :return: optional :class:`~aiida.engine.processes.workchains.utils.ProcessHandlerReport` instance to signal
+            that a problem was detected and potentially handled.
+        """
+        if node.exit_status == ArithmeticAddCalculation.exit_codes.ERROR_NEGATIVE_NUMBER.status:
+            self.ctx.inputs['x'] = orm.Int(abs(node.inputs.x.value))
+            self.ctx.inputs['y'] = orm.Int(abs(node.inputs.y.value))
+            return ProcessHandlerReport()
```

Comment:
I think this does not make it super clear that the handler should be a classmethod and so be defined within the class. I would vote to keep the context of the class definition and the indentation

Reply:
This was done to align better with the previous snippets.

But I see no harm in making this particular snippet more explicit.